application/src/main/java/org/togetherjava/tjbot/features/basic/MemberCounter.java

```diff
@@ -0,0 +1,48 @@
+package org.togetherjava.tjbot.features.basic;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.channel.concrete.Category;
+
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.features.Routine;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static java.lang.Integer.parseInt;
+
+public class MemberCounter implements Routine {
+    private final String wordsInCategory;
+
+    private final String updateMemberCountEveryHour;
+
+    public MemberCounter(Config config) {
+        this.wordsInCategory = config.getMemberCountCategoryName();
+        this.updateMemberCountEveryHour = config.getupdateMemberCountEveryHour();
+    }
+
+    private void updateCategoryName(Category category) {
+        int totalMemberCount = category.getGuild().getMemberCount();
+        String basename = category.getName();
+        category.getManager()
+            .setName("%s - %d Members".formatted(basename, totalMemberCount))
```

Comment:
this looks to me as if it executes twice the name will be:
Foo - 123 Members - 125 Members
and grow longer and longer each time. it likely needs to cut out the `- %d Member` suffix from the name first

Reply:
you sure? it is using `setName()`

Reply:
Hes correct i tested it out it did do it twice.

Reply:
ah yeah I see, its because you use the basename

Reply:
now you still add the ` -` part multiple times. id suggest u to use regex for this, also more readable.
```java
Pattern.compile("(.+) - \\d+ Members")
```
and then extract `group(1)` after matching.